targetFunctions/+domainsTF/customLayers.m

```diff
@@ -0,0 +1,201 @@
+    dataPresent, nParams, params, ~, resample, contrastBackgroundsType, cCustFiles, useImaginary] = extractProblemParams(problemStruct);
+    
+    calcSld = controls.calcSldDuringFit;
+    parallel = controls.parallel;
+    resamPars = controls.resamPars;
+             
+    % Pre-Allocation of output arrays...
+    backgroundParams = zeros(numberOfContrasts,1);
+    qzshifts = zeros(numberOfContrasts,1);
+    scalefactors = zeros(numberOfContrasts,1);
+    bulkIns = zeros(numberOfContrasts,1);
+    bulkOuts = zeros(numberOfContrasts,1);
+    resolutionParams = zeros(numberOfContrasts,1);
+    allRoughs = zeros(numberOfContrasts,1);
+    outSsubs = zeros(numberOfContrasts,1);
+    chis = zeros(numberOfContrasts,1);
+    domainLayerSlds = cell(numberOfContrasts,2);
+    domainSldProfiles = cell(numberOfContrasts,2);
+    shiftedData = cell(numberOfContrasts,1);
+    
+    reflectivity = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        reflectivity{i} = [1 1 ; 1 1];
+    end
+    
+    simulation = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        simulation{i} = [1 1 ; 1 1];
+    end
+    
+    domainAllLayers = cell(numberOfContrasts,2);
+    for i = 1:numberOfContrasts
+        domainAllLayers{i,1} = [1 1 1; 1 1 1];
+        domainAllLayers{i,2} = [1 1 1; 1 1 1];
+    end
+
+    calcAllLayers = cell(numberOfContrasts,2);
+    for i = 1:numberOfContrasts
+        calcAllLayers{i,1} = [1 ; 1];
+        calcAllLayers{i,2} = [1 ; 1];
+    end
+    
+    calcAllLayers1 = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        calcAllLayers1{i} = [1 ; 1];
+    end
+    
+    calcAllLayers2 = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        calcAllLayers2{i} = [1 ; 1];
+    end
+
+    sldProfiles = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        sldProfiles{i} = {[1 1 ; 1 1],[1 1 ; 1 1]};
+    end
+        
+    allLayers = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        allLayers{i} = {[1 1 1;1 1 1],[1 1 1;1 1 1]};
+    end
+    
+    layerSlds = cell(numberOfContrasts,1);
+    for i = 1:numberOfContrasts
+        layerSlds{i} = {[1 1 1;1 1 1],[1 1 1;1 1 1]};
+    end
+    
+    % Process the custom models....
+    [calcAllLayers,allRoughs] = domainsTF.customLayers.processCustomFunction(contrastBulkInIndices,contrastBulkOutIndices,...
+        bulkInArray,bulkOutArray,cCustFiles,numberOfContrasts,customFiles,params,useImaginary);
+    
+    for i = 1:size(calcAllLayers,1)
+        calcAllLayers1{i} = calcAllLayers{i,1};
+        calcAllLayers2{i} = calcAllLayers{i,2};
+    end
+    
+    if strcmpi(parallel, coderEnums.parallelOptions.Contrasts)
+    
+        % Parallel over all contrasts
+        parfor i = 1:numberOfContrasts
+        
+            [backgroundParams(i),qzshifts(i),scalefactors(i),bulkIns(i),...
```

Comment:
Is there a better way to align these functions for readability?

Reply:
Good question. I'm not sure. You could put a space between each variable, but they're already very long.

Reply:
Hmm, I think if we put the second square bracket on a new line that helps. Give me a mo.

Reply:
How does that look? I think it's a bit better.

Reply:
Yes it better, a bit easier to find the function name